src/Components/HomePage.tsx

```diff
@@ -27,161 +27,161 @@ import { logoutFromFirebase } from '../dataBaseUtils/auth/authUtils';
+              onClick={toggleDrawer}
+              sx={{
+                marginRight: '36px',
+                ...(open && { display: 'none' }),
+              }}
+            >
+              <MenuIcon />
+            </IconButton>
+            <Typography
+              component="h1"
+              variant="h6"
+              color="inherit"
+              noWrap
+              sx={{ flexGrow: 1 }}
+            >
+              Dashboard
+            </Typography>
+            {/* <IconButton color="inherit">
 							<Badge badgeContent={4} color="secondary">
 								<NotificationsIcon />
 							</Badge>
 						</IconButton> */}
-						<Button variant="contained" onClick={handleLogoutRequest}>Logout</Button>
-					</Toolbar>
-				</AppBar>
-				<Drawer variant="permanent" open={open}>
-					<Toolbar
-						sx={{
-							display: 'flex',
-							alignItems: 'center',
-							justifyContent: 'flex-end',
-							px: [1],
-						}}
-					>
-						<IconButton onClick={toggleDrawer}>
-							<ChevronLeftIcon />
-						</IconButton>
-					</Toolbar>
-					<Divider />
-					<List component="nav">
-						<MainListItems setActivePage={setActivePage} />
-						<Divider sx={{ my: 1 }} />
-						{/* {secondaryListItems} */}
-					</List>
-				</Drawer>
-				<Box
-					component="main"
-					sx={{
-						backgroundColor: (theme) =>
-							theme.palette.mode === 'light'
-								? theme.palette.grey[100]
-								: theme.palette.grey[900],
-						flexGrow: 1,
-						height: '100vh',
-						overflow: 'auto',
-					}}
-				>
-					<Toolbar />
-					{activePage === "invoices" && <InvoicesPage />}
-					{activePage === "dashboard" && <Dashboard />}
-					{activePage === "profile" && <Profile />}
-					{activePage === "allocations" && <Allocations />}
-				</Box>
-			</Box>
-		</ThemeProvider>
-	);
+            <Button variant="contained" onClick={handleLogoutRequest}>Logout</Button>
+          </Toolbar>
+        </AppBar>
+        <Drawer variant="permanent" open={open}>
+          <Toolbar
+            sx={{
+              display: 'flex',
+              alignItems: 'center',
+              justifyContent: 'flex-end',
+              px: [1],
+            }}
+          >
+            <IconButton onClick={toggleDrawer}>
+              <ChevronLeftIcon />
+            </IconButton>
+          </Toolbar>
+          <Divider />
+          <List component="nav">
+            <MainListItems setActivePage={setActivePage} />
+            <Divider sx={{ my: 1 }} />
+            {/* {secondaryListItems} */}
+          </List>
+        </Drawer>
+        <Box
+          component="main"
+          sx={{
+            backgroundColor: (theme) =>
+              theme.palette.mode === 'light'
+                ? theme.palette.grey[100]
+                : theme.palette.grey[900],
+            flexGrow: 1,
+            height: '100vh',
+            overflow: 'auto',
+          }}
+        >
+          <Toolbar />
+          {activePage === "invoices" && <InvoicesPage />}
```

Comment:
Here you can use the same logic as I showed in the `App` component - use a type and a component map.